src/main/java/duke/utilities/Ui.java

```diff
@@ -0,0 +1,145 @@
+package duke.utilities;
+
+import duke.task.Task;
+import duke.task.TaskList;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * A class that provides a simple user interface for the Duke system.
+ */
+public class Ui {
+
+    private static String line = "--------------------";
```

Comment:
One suggestion for your UI class is to abstract out the string values (perhaps enum class can even be used) and make the variables final as well as follow the naming convention for constants

for instance you can change

> private static String line = "--------------------";

to

> private static final String LINE = "--------------------";